src/Filters/GroupFilter.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace CodeIgniter\Shield\Filters;
+
+use CodeIgniter\Filters\FilterInterface;
+use CodeIgniter\HTTP\RequestInterface;
+use CodeIgniter\HTTP\Response;
+use CodeIgniter\HTTP\ResponseInterface;
+
+use CodeIgniter\Shield\Exceptions\GroupException;
+
+class GroupFilter implements FilterInterface
+{
+    /**
+     * Do whatever processing this filter needs to do.
+     * By default it should not return anything during
+     * normal execution. However, when an abnormal state
+     * is found, it should return an instance of
+     * CodeIgniter\HTTP\Response. If it does, script
+     * execution will end and that Response will be
+     * sent back to the client, allowing for error pages,
+     * redirects, etc.
+     *
+     * @param array|null $arguments
+     *
+     * @return mixed
+     */
+    public function before(RequestInterface $request, $arguments = null)
+    {
+        if (empty($arguments)) {
+            return;
+        }
+
+        if (! function_exists('auth')) {
+            helper('auth');
+        }
+
+        if (! auth()->loggedIn()) {
+            return redirect()->to('login');
+        }
+
+        foreach ($arguments as $group) {
+            if (auth()->user()->inGroup($group)) {
+                return;
+            }
+        }
```

Comment:
You should be able to simplify this: 

```suggestion
        if (auth()->user()->inGroup(...$arguments)) {
            return;
        }
```